spec/services/datetime_parser_spec.rb

```diff
@@ -92,4 +92,50 @@
       end
     end
   end
+
+  describe "#issues_for" do
+    let(:date_issues) { parser.issues_for(:schedule_date) }
+    let(:time_issues) { parser.issues_for(:schedule_time) }
+
+    context "an invalid time is provided" do
+      let(:parser) { DatetimeParser.new(date: valid_date, time: nil) }
+
+      it "finds issues for schedule_time" do
+        parser.parse
+
+        expect(date_issues.count).to eq(0)
```

Comment:
This line isn't needed as that's not what's being tested. I think this test is supposed to check that the `:schedule_time` issues are being returned, regardless of if there are any `: schedule_date` issues.

Reply:
Have removed from this test and created a dedicated test to ensure issues aren't falsely reported.